llvm/include/llvm/IR/CmpPredicate.h

```diff
@@ -53,6 +53,12 @@ class CmpPredicate {
   static std::optional<CmpPredicate> getMatching(CmpPredicate A,
                                                  CmpPredicate B);
 
+  /// Attempts to return a signed CmpInst::Predicate from the CmpPredicate. If
+  /// the CmpPredicate has samesign, return ICmpInst::getSignedPredicate,
+  /// dropping samesign information. Otherwise, return the predicate, dropping
+  /// samesign information.
+  CmpInst::Predicate getSignedPredicate() const;
```

Comment:
I'm a bit concerned this will cause confusion because people think it works like `ICmpInst::getSignedPredicate()`, i.e. always return signed predicate.

I'm not sure what a good name for this to clarify the difference. Maybe `getPreferredSignedPredicate()`?